src/overlays/actors/ovl_En_Daiku2/z_en_daiku2.c

```diff
@@ -22,56 +42,525 @@ const ActorInit En_Daiku2_InitVars = {
+        return;
+    }
+
+    if ((this->unk_28A != 5) && (sp98 != 2) && func_80BE64C0(this, globalCtx)) {
+        func_80BE6CFC(this);
+        return;
+    }
+
+    func_800B8614(&this->actor, globalCtx, 80.0f);
+    if ((this->unk_276 == 8) && (func_801378B8(&this->skelAnime, 6.0f) != 0)) {
+        Audio_PlayActorSound2(&this->actor, 0x2852);
+
+        for (i = 0; i < 10; i++) {
+            Math_Vec3f_Copy(&sp70, &this->actor.world.pos);
+
+            sp70.x += Math_SinS(this->actor.world.rot.y) * 70.0f;
+            sp70.y = sp70.y + 40.0f;
+            sp70.z += Math_CosS(this->actor.world.rot.y) * 70.0f;
+
+            sp70.x += randPlusMinusPoint5Scaled(5.0f);
+            sp70.y += randPlusMinusPoint5Scaled(5.0f);
+            sp70.z += randPlusMinusPoint5Scaled(5.0f);
+
+            sp88.y = -1.0f;
+
+            sp7C.x = (Rand_ZeroOne() - 0.5f) * 4.0f;
+            sp7C.y = (Rand_ZeroOne() * 10.0f) + 4.0f;
+            sp7C.z = (Rand_ZeroOne() - 0.5f) * 4.0f;
+
+            func_80BE7504(this, &sp70, &sp7C, &sp88, 0.01f - (Rand_ZeroFloat(1.0f) * 0.005f), 0x28);
+        }
+    }
+
+    temp_v0 = ABS_ALT((s16)(this->actor.yawTowardsPlayer - this->actor.home.rot.y));
+    if (temp_v0 >= 0x2891) {
+        if (sp98 != 2) {
+            if ((this->unk_276 == 5) || (this->unk_276 == 9)) {
+                func_80BE6408(this, 10);
+            } else if ((this->unk_276 == 10) && (this->unk_284 <= sp9C)) {
+                this->unk_284 = SkelAnime_GetFrameCount(&D_06002134.common);
+                SkelAnime_ChangeAnim(&this->skelAnime, &D_06002134, -1.0f, this->unk_284, 0.0f, 2, -4.0f);
+                this->unk_276 = 11;
+            } else if ((this->unk_276 == 11) && (sp9C <= 0.0f)) {
+                func_80BE6408(this, 8);
+            }
+        }
+    } else if (sp98 != 2) {
+        if ((this->unk_276 != 9) && (this->unk_276 != 10)) {
+            func_80BE6408(this, 9);
+        } else if ((this->unk_276 == 9) && (this->unk_284 <= sp9C)) {
+            func_80BE6408(this, 10);
+        }
+    }
+}
+
+void func_80BE6B40(EnDaiku2* this, GlobalContext* globalCtx) {
+    s32 day = gSaveContext.day;
+
+    this->unk_288 = 1;
+    if ((day != 3) && Flags_GetSwitch(globalCtx, this->unk_278)) {
+        this->actionFunc = func_80BE6BC0;
+    } else {
+        func_80BE6408(this, 5);
+        this->actionFunc = func_80BE6BC0;
+    }
+}
+
+void func_80BE6BC0(EnDaiku2* this, GlobalContext* globalCtx) {
+    s32 day;
+
+    Math_SmoothStepToS(&this->actor.world.rot.y, this->actor.yawTowardsPlayer, 1, 0xBB8, 0);
+    if ((func_80152498(&globalCtx->msgCtx) == 5) && (func_80147624(globalCtx) != 0)) {
+        day = gSaveContext.day - 1;
+        func_801477B4(globalCtx);
+
+        if (this->unk_288 == 2) {
+            this->actionFunc = func_80BE6D40;
+        } else if (this->unk_288 == 3) {
+            this->actionFunc = func_80BE6EF0;
+        } else if ((this->unk_28A == 0) || (this->unk_28A == 2)) {
+            this->unk_28A += 1;
```

Comment:
Feel free to ignore this, I just personally feel it's more readable

```suggestion
            this->unk_28A++;
```